src/NuGet.Services.AzureSearch/SearchService/StatusService.cs

```diff
@@ -0,0 +1,145 @@
+                    response,
+                    "getting server information"));
+
+            return response;
+        }
+
+        private async Task TryAsync<T>(
+            Func<Task<T>> getAsync,
+            StatusResponse response,
+            string operation)
+        {
+            try
+            {
+                await Task.Yield();
+                await getAsync();
+            }
+            catch (Exception ex)
+            {
+                response.Success = false;
+                _logger.LogError(0, ex, "When getting the server status, {Operation} failed.", operation);
+            }
+        }
+
+        private Task<ServerStatus> GetServerStatusAsync(Assembly assembly)
+        {
+            DateTimeOffset processStartTime;
+            int processId;
+            using (var process = Process.GetCurrentProcess())
+            {
+                processStartTime = process.StartTime.ToUniversalTime();
+                processId = process.Id;
+            }
+
+            var serverStatus = new ServerStatus
+            {
+                AssemblyBuildDateUtc = GetAssemblyMetadataOrNull(assembly, "BuildDateUtc"),
+                AssemblyCommitId = GetAssemblyMetadataOrNull(assembly, "CommitId"),
+                AssemblyInformationalVersion = GetAssemblyInformationalVersionOrNull(assembly),
+                DeploymentLabel = _options.Value.DeploymentLabel,
+                MachineName = Environment.MachineName,
+                InstanceId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID"),
+                ProcessDuration = DateTimeOffset.UtcNow - processStartTime,
+                ProcessId = processId,
+                ProcessStartTime = processStartTime,
+            };
+
+            return Task.FromResult(serverStatus);
+        }
+
+        private async Task<AuxiliaryFilesMetadata> GetAuxiliaryFilesMetadataAsync()
+        {
+            await _auxiliaryDataCache.InitializeAsync();
+            return _auxiliaryDataCache.Get().Metadata;
+        }
+
+        private static string GetAssemblyInformationalVersionOrNull(Assembly assembly)
+        {
+            return assembly
+                .GetCustomAttributes<AssemblyInformationalVersionAttribute>()
+                .Select(x => x.InformationalVersion)
+                .FirstOrDefault();
+        }
+
+        private static string GetAssemblyMetadataOrNull(Assembly assembly, string name)
+        {
+            return assembly
+                .GetCustomAttributes<AssemblyMetadataAttribute>()
+                .Where(x => x.Key == name)
+                .Select(x => x.Value)
+                .FirstOrDefault();
+        }
+
+        private static async Task<IndexStatus> GetIndexStatusAsync(ISearchIndexClientWrapper index)
+        {
+            var documentCount = await index.Documents.CountAsync();
+            var warmQueryDuration = await Measure.DurationAsync(() => index.Documents.SearchAsync("*", new SearchParameters()));
```

Comment:
Will this query get more expensive as more documents are added? Will this method executed often, could be any server impact?

Reply:
I found no indication from documentation that it will become _significantly_ slower. The index I tested it on had full production data in it and was running on the minimum replica/partition count that we would go live with.

If this query is slow we will bigger problems. This is essentially the default query that is done when someone opens the [default packages list on gallery](https://www.nuget.org/packages) or on client.

Reply:
By the way, this is mitigated by https://github.com/nuget/nugetgallery/issues/6463.